### PR TITLE
Variable.Set Should Invalidated the Cached Value

### PR DESCRIPTION
**Pull request by antJack**

### CODE REVIEW

File: variable/api.go [hunk 1]
```diff
@@ -182,6 +182,9 @@ func setFlushedValue(ctx context.Context, index uint32, value interface{}) error
 			if setter == nil {
 				return errors.New(errSetterNotFound + variable.Name())
 			}
+
+			// should invalidate the cached value before setting it to a new one
+			variableValue.Valid = false
```
Comment (taoyuanyuan) [inline variable/api.go:187]: What scenarios will lead to failure? I see `DefaultSetter` do not judge `Valid`.

**Comment by nejisama** [comment]
之前的改动中，简化了缓存flag的设置
https://github.com/mosn/pkg/commit/4d6d91026cb9716165e9d26d8a67cdd657d96b9d
导致存在这个问题

File: variable/api.go (hunk 1 above)
Comment (antJack) [reply variable/api.go:187]: if we both have getter and setter, then the cached value will not be modified by setter and may return obsolete value

**Comment by nejisama** [comment]
see #75

**Review by taoyuanyuan — APPROVED** [review]
LGTM

**Review by nejisama — APPROVED** [review]

**Comment by nejisama** [comment]
> see #75

this one is better

**Pull request merged**